lib/collision_detection.dart

```diff
@@ -0,0 +1,32 @@
+import 'dart:math' as math;
+
+import 'extensions/vector2.dart';
+
+/// Checks whether the [polygon] represented by the list of [Vector2] contains
+/// the [point].
+bool containsPoint(Vector2 point, List<Vector2> polygon) {
```

Comment:
@spydon I think I have some of these methods on flame_geom, have you taken a look at that?

Reply:
I only think `flame_geom` can handle basic shapes and it isn't migrated to `Vector2` yet, I can have a look at migrating it once I'm done with the follow up collision detection PR that is coming after this one.
And if we do the migration, we should really rename `flame_geom`, I really dislike abbreviations in package names. 
😅